### ISSUE TITLE
unicode nicknames

### ISSUE THREAD

**Issue by graydon** [mozilla/MozStumbler#568]
We had a report of non-ASCII unicode nicknames (particularly the name `caméléon`) not making it from the stumbler all the way through to ichnaea's leaderboard (bug https://github.com/mozilla/ichnaea/issues/160 ). We've confirmed (see screenshot in https://github.com/mozilla/ichnaea/pull/185 ) that we can synthesize, submit, store and display such strings, so our next guess (unless it's some part of our hosting / proxying pipeline) is that maybe the stumbler is losing them somehow or failing to submit them.

Any digging you can do on this topic appreciated.

**Comment by Djfe** [comment]
I don't know if this is relevant but why is .trim() used on the nickname while getting it? Would be better to do this while setting/saving it, I think.

Something that could be fixed together with this issue:
Nicknames go from 2 up to 30 (or was it 32?) characters.
The nickname edit should check if the Nicknames length is in this valid interval or else return an alert or something.

**Comment by hannosch** [comment]
@Djfe mentioned on the server side issue, that HTTP headers don't actually support utf-8 encoded data. They are only defined as ASCII, so anything in between the stumbler app and the server side could break things. We need to come up with an ASCII-compatible encoding for the header.

**Comment by Djfe** [comment]
@crankycoder was this fixed by #890, too?

**Comment by Djfe** [comment]
can't we convert Unicode nicknames and e-mails into ASCII by writing \u1234
while \u is the indication for a unicode char?

**Comment by crankycoder** [comment]
Blargh.  We do encode nicknames as UTF-8.  I'd rather not deal with this problem in the client code right now as we've got a fix by simply not doing nicknames at all in any of the stumbler client or server side code.  

Instead - we delegate all the unicode/nickname problems to Firefox Accounts now.

That code went in on 916db01d8538db94c9b968387341bf56824d58c5 but we're going to delay release of FxA logins until the new leaderboard is out at the end of 2015Q3.

I've migrated this bug over to https://github.com/JaredKerim-Mozilla/leaderboard-server/issues/7

I think the main risk here is rendering non-western glyphs on the front end UI.
